### PR TITLE
GTESTS: add send IOVs testing in test_ucp_stream

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Do not merge before #1874

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4720/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4843/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4844/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4845/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_stream.cc [hunk 1]
```diff
@@ -26,17 +26,47 @@ class test_ucp_stream : public ucp_test {
...
+            buf = tmp;
+            count = i;
+            break;
+        case UCP_DATATYPE_IOV:
+            buf = iov_;
+            count = iov_cnt_;
+            break;
+        }
+
+        ASSERT_TRUE(buf && count);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_stream.cc]: need to separate to 2 statements

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4866/ for details (Mellanox internal link).

**Pull request merged**